<docs>
	<doc for="ArrayTypeMismatchException.ArrayTypeMismatchException3">
		<summary>
			<para>Initializes a new instance of the <see cref="System.ArrayTypeMismatchException"/> class with serialized data.</para>
		</summary>
		<param name="info">The object that holds the serialized object data.</param>
		<param name="context">The contextual information about the source or destination.</param>
		<remarks>
			<para>This constructor is called during deserialization to
      reconstitute the exception object transmitted over a stream. For more
      information, see <see topic="cpconSerialization"/>.</para>
		</remarks>
		<seealso topic="cpconSerialization"/>
	</doc><doc for="ArrayTypeMismatchException.ArrayTypeMismatchException">
		<overload>
			<para>Initializes a new instance of the
   <see cref="System.ArrayTypeMismatchException"/> class.</para>
		</overload>
		<summary>
			<para>Initializes a new instance of the
   <see cref="System.ArrayTypeMismatchException"/> class.</para>
		</summary>
		<remarks>
			<para>This constructor initializes the <see cref="System.Exception.Message"/>
property of the new instance to a system-supplied message that describes the
error, such as "Source array type cannot be assigned to destination array type."
This message takes into account the current system culture.</para>
			<para>The following table shows the initial property values for
   an instance of <see cref="System.ArrayTypeMismatchException"/>.</para>
			<list type="table"><listheader>
					<term>Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description>A null reference (<see langword="Nothing"/> in
Visual Basic).</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The localized error message string.</description>
				</item>
			</list>
		</remarks>
		<example>
			<codelink SampleID="ArrayTypeMisMatch_Constructor1" SnippetID="1"/>
		</example>
	</doc><doc for="ArrayTypeMismatchException.ArrayTypeMismatchException1">
		<summary>
			<para>Initializes a new instance of the
   <see cref="System.ArrayTypeMismatchException"/>
   class with a specified error message.</para>
		</summary>
		<param name="message">A <see cref="System.String"/> that describes the error.</param>
		<remarks>
			<para>The content of the <paramref name="message"/> parameter is
   intended to be understood by humans. The caller of this constructor is required
   to ensure that this string has been localized for the current system
   culture.</para>
			<para>The following table shows the initial property values
   for an instance of <see cref="System.ArrayTypeMismatchException"/>.</para>
			<list type="table"><listheader>
					<term>Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description>A null reference (<see langword="Nothing"/> in
Visual Basic).</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The error message string.</description>
				</item>
			</list>
		</remarks>
		<example>
			<codelink SampleID="ArrayTypeMisMatch_Constructor2" SnippetID="1"/>
		</example>
	</doc><doc for="ArrayTypeMismatchException.ArrayTypeMismatchException2">
		<summary>
			<para>Initializes a new instance of the <see cref="System.ArrayTypeMismatchException"/>
class with a specified error message and a reference to the
inner exception that is the cause of this exception.</para>
		</summary>
		<param name="message">The error message that explains the reason for the exception.</param>
		<param name="innerException">The exception that is the cause of the current exception. If the <paramref name="innerException"/> parameter is not a null reference, the current exception is raised in a <see langword="catch"/> block that handles the inner exception.</param>
		<remarks>
			<para>An exception that is thrown as a direct result of a
      previous exception should include a reference to the previous exception in
      the <see cref="System.Exception.InnerException"/> property. The
   <see cref="System.Exception.InnerException"/> property returns
      the same value that is passed into the constructor, or a null reference
      (<see langword="Nothing"/> in Visual Basic) if the <see cref="System.Exception.InnerException"/>
      property does not supply the inner exception value to
      the constructor. </para>
			<para>The following table shows the initial property values for an instance of
   <see cref="System.ArrayTypeMismatchException"/>.</para>
			<list type="table"><listheader>
					<term>Property</term>
					<description>Value</description>
				</listheader>
				<item>
					<term><see cref="System.Exception.InnerException"/></term>
					<description>The inner exception reference.</description>
				</item>
				<item>
					<term><see cref="System.Exception.Message"/></term>
					<description>The error message string.</description>
				</item>
			</list>
		</remarks>
		<example>
			<codelink SampleID="ArrayTypeMisMatch_Constructor3" SnippetID="1"/>
		</example>
		<seealso cref="System.Exception"/>
		<seealso topic="cpconHandlingThrowingExceptions"/>
	</doc><doc for="ArrayTypeMismatchException">
		<summary>
			<para>The exception that is thrown when an attempt is made to
      store an element of the wrong type within an array.</para>
		</summary>
		<remarks>
			<para><see cref="System.ArrayTypeMismatchException"/> is thrown when the system cannot
   convert the element to the type declared for the array. For example, an element
   of type <see cref="System.String"/> cannot be stored in an <see cref="System.Int32"/> array because conversion between these
   types is not supported. It is generally unnecessary for applications to throw
   this exception. </para>
			<para> The following Microsoft intermediate language (MSIL) instructions throw <see cref="System.ArrayTypeMismatchException"/>:</para>
			<list type="bullet"><item>
					<term>
      ldelem.&lt;type&gt;</term>
				</item>
				<item>
					<term>
      ldelema</term>
				</item>
				<item>
					<term>
      stelem.&lt;type&gt;</term>
				</item>
			</list>
			<para><see cref="System.ArrayTypeMismatchException"/> uses the HRESULT COR_E_ARRAYTYPEMISMATCH, which has the
value 0x80131503.</para>
			<para>For a list of initial property values for an instance of <see cref="System.ArrayTypeMismatchException"/>, see the <see cref="System.ArrayTypeMismatchException"/> constructors.</para>
		</remarks>
		<example>
			<codelink SampleID="System.ArrayTypeMismatch" SnippetID="1"/>
		</example>
		<seealso cref="System.Exception"/>
		<seealso topic="cpconHandlingThrowingExceptions"/>
	</doc>
</docs>
